### PR TITLE
Update go deps

### PR DESCRIPTION
**Pull request by stlava**
Catch up to latest dependencies

- Updated `aws` sdk
- Updated `datadog` client
- Updated `backoff`/retry module
- Updated `logrus` module
- Updated postgres replication client
- Updated the CPU and memory profiles to match new libraries

### CODE REVIEW

File: replication/client/client.go [hunk 1]
```diff
@@ -332,6 +333,7 @@ func (c *Replicator) Start(progressChan <-chan uint64) {
 				continue
 			}
 
+			fmt.Println("returning")
```
Comment (ruban-rengaraju) [inline replication/client/client.go:336]: update this to a log (or if this is for test, remove this)

File: itests/tests/kinesis/test_strict_order_shared/envfile.env [hunk 2]
```diff
@@ -4,4 +4,5 @@ WORKERS=2
 PARTITION_METHOD=transaction-bucket
 BATCHER_ROUTING_METHOD=partition
 MEMPROFILE=/perf/mem.pprof
-CPUPROFILE=/perf/cpu.pprof
\ No newline at end of file
+CPUPROFILE=/perf/cpu.pprof
+GODEBUG=memprofilerate=1
```
Comment (ruban-rengaraju) [inline itests/tests/kinesis/test_strict_order_shared/envfile.env:8]: nit, newline

**Review by ruban-rengaraju — CHANGES_REQUESTED** [review]
left one small request

File: replication/client/client.go (hunk 1 above)
Comment (stlava) [reply replication/client/client.go:336]: Oops left that during debugging.

**Pull request merged**